druid-shell/src/windows/mod.rs

```diff
@@ -240,18 +253,31 @@ impl MyWndProc {
             println!("EndDraw error: {:?}", e);
         }
         if anim {
-            let handle = self.handle.borrow().get_idle_handle().unwrap();
+            let handle = handle.borrow().get_idle_handle().unwrap();
             // Note: maybe add WindowHandle as arg to idle handler so we don't need this.
             let handle2 = handle.clone();
             handle.add_idle(move |_| handle2.invalidate());
         }
     }
 }
 
+impl MyWndProc {
+    /// Create debugging output for dropped messages due to wndproc reeentrancy.
+    ///
+    /// In the future, we choose to do something else other than logging and dropping,
+    /// such as queuing and replaying after the nested call returns.
+    fn log_dropped_msg(&self, hwnd: HWND, msg: UINT, wparam: WPARAM, lparam: LPARAM) {
+        eprintln!(
+            "dropped message 0x{:x}, hwnd={:?}, wparam=0x{:x}, lparam=0x{:x}",
+            msg, hwnd, wparam, lparam
+        );
+    }
+}
+
 impl WndProc for MyWndProc {
-    fn connect(&self, handle: &WindowHandle, state: WndState) {
+    fn connect(&self, handle: &WindowHandle, mut state: WndState) {
```

Comment:
So just to understand, doing anything before calling `connect` will be a crash?

Reply:
Anything before `connect` is considered part of the construction of the struct. This is basically just a way to get a window handle into MyWndProc - for reasons that are not very interesting it needs to be created before that window handle is available, though I wouldn't be surprised if there was a way to change the order up a bit. It'll be a crash if it tries to do anything requiring the handle. I think that's not exposed to client code though.